.github/workflows/purge-test-resources.yaml

```diff
@@ -105,7 +105,7 @@ jobs:
     name: Create issue for failing purge test resources run
     needs: [purge_ghcr_dev, purge_azure_resources]
     runs-on: ubuntu-latest
-    if: always() && ${{ github.event_name != 'workflow_dispatch' }} && (needs.purge_ghcr_dev.result == 'failure' || needs.purge_azure_resources.result == 'failure')
+    if: always() && (${{ github.event_name != 'workflow_dispatch' }} && needs.purge_ghcr_dev.result == 'failure' || needs.purge_azure_resources.result == 'failure')
```

Comment:
I think `always()` needs to be removed here. Could you check the docs for the `always()` keyword?

Reply:
I don't think `always()` was the issue since it was creating issues only on failures before. I think it was just the syntax of the conditional that was causing issues. These are the docs for `always()`: https://docs.github.com/en/actions/learn-github-actions/expressions#always. So it's used when we want this to run even if a job is cancelled

Reply:
Could you explain to me the difference between the conditions here? I might be missing something here.

Reply:
I think this should have been `always() && (${{ github.event_name != 'workflow_dispatch' }} && (needs.purge_ghcr_dev.result == 'failure' || needs.purge_azure_resources.result == 'failure'))`. But I did some more testing and it looks like this conditional might be causing issues because of how `always()` works on GH actions.

Unless I'm missing something, the conditional evaluated to `true&false` before and now it's evaluating to `true&true&false` which should still be false. so I'm actually not sure why it was working correctly before but failing now. Going to try with removing the `always()` as suggested and seeing if that fixes it